README
About
=====

I'm making a stock market game. It's open source and licensed under a Creative Commons Attribution 3.0 Unported License.

Main.java is the only file at the moment because I haven't seen the need for more (yet). Over the next month the stock market simulator will have most of its features done and hopefully the gui (but don't count on the gui part).


Recent News
===========

I added a very early version of the GUI so people can see how I'm going to do it (to some degree). If you would like to run it with the one-quarter-finished gui and you don't want to look at the code, run it at the command line like this: "java -jar stock-market-v0.5.jar showgui"


Version Changes
===============

For more detailed changes, see changelog.txt

Changes this version (v0.4):
 + Added more companies
 + Added two more rumors
 + Better fluctuations
 + Company traits (i.e. steady or unsteady)
 + Added chance of buy backs
 * Made it so players can't by stock worth $0.00 for obvious reasons
 * Made it easier to add more companies to the exchange
 * Tweaked how share prices are managed (now done with an array)
 * Split code into three classes
 * Fixed buyStock so that it actually changes how much money the player has
 * Fixed buyStock and sellStock so you can go back to the main menu
 * Fixed bug that doesn't show rumors when viewing the main menu more than once per day
 * Modified myShares to only show companies that the player has shares in
 * Modified how stock prices are listed

Features in next version (v0.5):
 * Some form of GUI

Features in future versions:
 * Possibility for companies to go completely bankrupt and not listed in the exchange
 * (Maybe) Dividends paid to the user every X days/months/years
 * User can save their progress and high score
 * More loan sharks with different intreats rates
 * More ways to lose
 * (More features as I think of them)

changelog.txt
Changelog

v0.1 (04/01/12)
 + Added companies
 + Added user money
 + Added simple share price fluctuation
 + Added days

v0.2 (04/01/12)
 + Users can buy/sell stock
 + Added pseudo-debug function to simulate the next 100 days (input 7 at the main menu)
 + Users can see how much money they have
 + Added "My Shares"
 * Made all prices into doubles
 * Tweaked fluctuation "algorithm"

v0.3 (05/01/12)
 + Added GFC event
 + Player can borrow money from loan shark
 + Players can now lose
 + Added game over screen
 * Rumors can now be false
 * Tweaked nextDay method lots to accommodate for new functions

v0.3.1 (05/01/12)
 * Fixed loanShark method not granting the player their money
 * Tweaked the main menu

v0.4 (05/01/12)
 + Added more companies
 + Added two more rumors
 + Better fluctuations
 + Company traits (i.e. steady or unsteady)
 + Added chance of buy backs
 * Made it so players can't by stock worth $0.00 for obvious reasons
 * Made it easier to add more companies to the exchange
 * Tweaked how share prices are managed (now done with an array)
 * Split code into three classes
 * Fixed buyStock so that it actually changes how much money the player has
 * Fixed buyStock and sellStock so you can go back to the main menu
 * Fixed bug that doesn't show rumors when viewing the main menu more than once per day
 * Modified myShares to only show companies that the player has shares in
 * Modified how stock prices are listed

src/stock/b2kan/com/GUI.java
package stock.b2kan.com;

import java.awt.ComponentOrientation;
import java.awt.Container;
import java.awt.GridBagConstraints;
import java.awt.GridBagLayout;
import java.awt.Insets;
import java.awt.event.ActionListener;
import java.awt.event.ActionEvent;

import javax.swing.JDialog;
import javax.swing.JFrame;
import javax.swing.JButton;
import javax.swing.JLabel;
import javax.swing.JOptionPane;
import javax.swing.JPanel;

@SuppressWarnings("serial")
public class GUI extends JFrame {
	private static JButton buyDerp, buyHerp, buyFoo, buyBar, buyNano, buyApple, nextDay;
	private static JLabel companyName, priceDerp, priceHerp, priceFoo, priceBar, priceNano, priceApple;
	public static GridBagConstraints con	= new GridBagConstraints();
	public static JFrame gui	= new JFrame("GUI tests");
	public static GUI content	= new GUI(gui);
	
	public static void showGUI() {
		gui.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
		content.getContentPane();
		
		gui.pack();
		gui.setVisible(true);
	}
	
	public GUI(Container pane) {
		pane.setLayout(new GridBagLayout());
		pane.setComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT);
		con.fill				= GridBagConstraints.HORIZONTAL;
		con.gridx	= 0;
		con.gridy	= 0;
		con.insets	= new Insets(10,10,10,10);
		JPanel buyShares	= new JPanel();
		buyShares.setLayout(new GridBagLayout());
		pane.add(buyShares, con);
		
		companyName		= new JLabel("Derp Inc"); 
		buyDerp			= new JButton("Buy shares");
		con.gridx	= 0;
		con.gridy	= 0;
		buyShares.add(companyName, con);
		con.gridx	= 1;
		buyShares.add(buyDerp, con);

		companyName		= new JLabel("Herp Industries"); 
		buyHerp			= new JButton("Buy shares");
		con.gridx	= 0;
		con.gridy	= 1;
		buyShares.add(companyName, con);
		con.gridx	= 1;
		buyShares.add(buyHerp, con);
		
		companyName		= new JLabel("Foo Corp"); 
		buyFoo			= new JButton("Buy shares");
		con.gridx	= 0;
		con.gridy	= 2;
		buyShares.add(companyName, con);
		con.gridx	= 1;
		buyShares.add(buyFoo, con);
		
		companyName		= new JLabel("Bar Pty Ltd"); 
		buyBar			= new JButton("Buy shares");
		con.gridx	= 0;
		con.gridy	= 3;
		buyShares.add(companyName, con);
		con.gridx	= 1;
		buyShares.add(buyBar, con);
		
		companyName		= new JLabel("Nanohard"); 
		buyNano			= new JButton("Buy shares");
		con.gridx	= 0;
		con.gridy	= 4;
		buyShares.add(companyName, con);
		con.gridx	= 1;
		buyShares.add(buyNano, con);
		
		companyName		= new JLabel("Applesoft"); 
		buyApple		= new JButton("Buy shares");
		con.gridx	= 0;
		con.gridy	= 5;
		buyShares.add(companyName, con);
		con.gridx	= 1;
		buyShares.add(buyApple, con);
		
		con.gridx	= 0;
		
		showPrices(pane, false);
		con.gridy	= Main.listed.length;
		nextDay		= new JButton("Next Day");
		buyShares.add(nextDay, con);
		
		HandlerClass handler	= new HandlerClass();
		buyDerp.addActionListener(handler);
		buyHerp.addActionListener(handler);
		buyFoo.addActionListener(handler);
		buyBar.addActionListener(handler);
		buyNano.addActionListener(handler);
		buyApple.addActionListener(handler);
		nextDay.addActionListener(handler);
	}
	
	public static void showPrices(Container pane, boolean update) {
		con.gridx	= 1;
		con.gridy	= 0;
		con.weighty	= 0.0;
		JPanel sharePrice	= new JPanel();
		sharePrice.setLayout(new GridBagLayout());
		pane.add(sharePrice, con);
		if(update == false) {
			con.gridx	= 0;
			con.gridy	= 0;
			companyName	= new JLabel(Main.listed[con.gridy]);
			sharePrice.add(companyName, con);
			
			con.gridx	= 1;
			String thisPrice	= Double.toString(Main.exchange[con.gridy]);
			priceDerp	= new JLabel("$"+thisPrice);
			sharePrice.add(priceDerp, con);
			
			con.gridx	= 0;
			con.gridy++;
			

			companyName	= new JLabel(Main.listed[con.gridy]);
			sharePrice.add(companyName, con);
			
			con.gridx	= 1;
			thisPrice	= Double.toString(Main.exchange[con.gridy]);
			priceHerp	= new JLabel("$"+thisPrice);
			sharePrice.add(priceHerp, con);
			
			con.gridx	= 0;
			con.gridy++;
			

			companyName	= new JLabel(Main.listed[con.gridy]);
			sharePrice.add(companyName, con);
			
			con.gridx	= 1;
			thisPrice	= Double.toString(Main.exchange[con.gridy]);
			priceFoo	= new JLabel("$"+thisPrice);
			sharePrice.add(priceFoo, con);
			
			con.gridx	= 0;
			con.gridy++;
			

			companyName	= new JLabel(Main.listed[con.gridy]);
			sharePrice.add(companyName, con);
			
			con.gridx	= 1;
			thisPrice	= Double.toString(Main.exchange[con.gridy]);
			priceBar	= new JLabel("$"+thisPrice);
			sharePrice.add(priceBar, con);
			
			con.gridx	= 0;
			con.gridy++;
			

			companyName	= new JLabel(Main.listed[con.gridy]);
			sharePrice.add(companyName, con);
			
			con.gridx	= 1;
			thisPrice	= Double.toString(Main.exchange[con.gridy]);
			priceNano	= new JLabel("$"+thisPrice);
			sharePrice.add(priceNano, con);
			
			con.gridx	= 0;
			con.gridy++;
			

			companyName	= new JLabel(Main.listed[con.gridy]);
			sharePrice.add(companyName, con);
			
			con.gridx	= 1;
			thisPrice	= Double.toString(Main.exchange[con.gridy]);
			priceApple	= new JLabel("$"+thisPrice);
			sharePrice.add(priceApple, con);
			
			con.gridx	= 0;
			con.gridy++;
		} else {
			int count	= 0;
			String thisPrice	= Double.toString(Main.exchange[count]);
			priceDerp.setText("$"+thisPrice);
			count++;
			thisPrice	= Double.toString(Main.exchange[count]);
			priceHerp.setText("$"+thisPrice);
			count++;
			thisPrice	= Double.toString(Main.exchange[count]);
			priceFoo.setText("$"+thisPrice);
			count++;
			thisPrice	= Double.toString(Main.exchange[count]);
			priceBar.setText("$"+thisPrice);
			count++;
			thisPrice	= Double.toString(Main.exchange[count]);
			priceNano.setText("$"+thisPrice);
			count++;
			thisPrice	= Double.toString(Main.exchange[count]);
			priceApple.setText("$"+thisPrice);
			pane.repaint();
		}
		//sharePrice.add(nextDay, con);
	}
	
	private class HandlerClass implements ActionListener {
		int amount;
		JOptionPane yesNo;
		Object obj;
		Object[] options	= new String[] {"Yes", "No"};
		JDialog dialog;
		public void actionPerformed(ActionEvent event) {
			if(event.getSource() == nextDay) {
				Main.nextDay();
				showPrices(gui, true);
				Game.showNews();
			}
			if(event.getSource() == buyDerp || event.getSource() == buyHerp || event.getSource() == buyFoo || event.getSource() == buyBar || event.getSource() == buyNano || event.getSource() == buyApple) {
				String StAmount	= JOptionPane.showInputDialog("How many shares would you like to buy?");
				amount		= Integer.parseInt(StAmount);
			}
			
			if(event.getSource() == buyDerp) {
				yesNo	= new JOptionPane("Buy "+amount+" shares in Derp Inc for $"+Main.exchange[0]+" each? (Total $"+(amount*Main.exchange[0])+")");
				yesNo.setOptions(options);
				dialog = yesNo.createDialog(new JFrame(), "Dialog");
				dialog.setVisible(true);
				obj	= yesNo.getValue();
				System.out.println("User chose: "+obj);
			}
			
			if(event.getSource() == buyHerp) {
				yesNo	= new JOptionPane("Buy "+amount+" shares in Herp Industries for $"+Main.exchange[1]+" each? (Total $"+(amount*Main.exchange[1])+")");
				yesNo.setOptions(options);
				dialog = yesNo.createDialog(new JFrame(), "Dialog");
				dialog.setVisible(true);
				obj	= yesNo.getValue();
				System.out.println("User chose: "+obj);
			}
			
			if(event.getSource() == buyFoo) {
				yesNo	= new JOptionPane("Buy "+amount+" shares in FooCorp for $"+Main.exchange[2]+" each? (Total $"+(amount*Main.exchange[2])+")");
				yesNo.setOptions(options);
				dialog = yesNo.createDialog(new JFrame(), "Dialog");
				dialog.setVisible(true);
				obj	= yesNo.getValue();
				System.out.println("User chose: "+obj);
			}
			
			if(event.getSource() == buyBar) {
				yesNo	= new JOptionPane("Buy "+amount+" shares in Bar Pty Ltd for $"+Main.exchange[3]+" each? (Total $"+(amount*Main.exchange[3])+")");
				yesNo.setOptions(options);
				dialog = yesNo.createDialog(new JFrame(), "Dialog");
				dialog.setVisible(true);
				obj	= yesNo.getValue();
				System.out.println("User chose: "+obj);
			}
			
			if(event.getSource() == buyNano) {
				yesNo	= new JOptionPane("Buy "+amount+" shares in Nanohard for $"+Main.exchange[4]+" each? (Total $"+(amount*Main.exchange[4])+")");
				yesNo.setOptions(options);
				dialog = yesNo.createDialog(new JFrame(), "Dialog");
				dialog.setVisible(true);
				obj	= yesNo.getValue();
				System.out.println("User chose: "+obj);
			}
			
			if(event.getSource() == buyApple) {
				yesNo	= new JOptionPane("Buy "+amount+" shares in Applesoft for $"+Main.exchange[5]+" each? (Total $"+(amount*Main.exchange[5])+")");
				yesNo.setOptions(options);
				dialog = yesNo.createDialog(new JFrame(), "Dialog");
				dialog.setVisible(true);
				obj	= yesNo.getValue();
				System.out.println("User chose: "+obj);
			}
		}
	}
	
}


src/stock/b2kan/com/Game.java
package stock.b2kan.com;

import java.util.Scanner;

public class Game {
	//private static int input		= 0;
	private static Scanner inputScan= new Scanner(System.in);
	
	public static void rollForRumor() {
		if(Main.rand.nextInt(6) == 1) {
			boolean loop	= true;
			int random	= Main.rand.nextInt(6);
			while(loop) {
				if(Main.rumors[random] == Main.oldRumor) {
					random	= Main.rand.nextInt(6);
				} else {
					if(random == 5) {
						Main.barBankrupt	= true;
					}
					Main.thisRumor	= Main.rumors[random];
					Main.rumor	= random;
					loop	= false;
				}
			}
			Main.oldRumor	= Main.thisRumor;
		}
		
	}
	
	public static void rollForGfc() {
		if(Main.rand.nextInt(100) == 1) {
			Main.gfc		= true;
			Main.rumor		= -2;
			Main.thisRumor	= "A global financial crisis is predicted! Expect share prices to severely drop!";
		}
	}
	
	public static void rollForBuyBack() {
		if(Main.rand.nextInt(100) == 1) {
			int random			= Main.rand.nextInt(6);
			double currentPrice	= Main.exchange[random];
			if(Main.inventory1[random] != 0) {
				System.out.println(Main.listed[random]+" are offering to buy back all of their shares at $"+(currentPrice + (currentPrice * .2))+", do you accept their offer? (y/n)");
				String yesno	= inputScan.next();
				
				if(yesno.equals("y") || yesno.equals("Y") || yesno.equals("yes")) {
					Main.inventory1[random]	= 0;
					Money.money	= Money.money + (Main.inventory1[random] * (currentPrice * .2));
				}
			}
		}
	}
	
	public static void showNews() {
		if(Main.rumor != -1 || Main.thisRumor == "The rumor was true!" || Main.thisRumor == "The rumor was false!") {
			System.out.println("\nLatest News: "+Main.thisRumor);
		}
		if(Main.news != "") {
			System.out.println("\n"+Main.news);
			Main.news	= "";
		}
	}
	
	public static void sleep(int time) {
		try {
			Thread.sleep(time);
		} catch (InterruptedException e) {
			e.printStackTrace();
		}
	}
	
	public static void gameOver(int reason) {
		System.out.println("\n\n\n\n\n\n\n\n\n" +
							"	GAME OVER\n");
		if(reason == 1) { // Reason 1 means they lost because they couldn't pay back Logan
			System.out.println("You failed to pay back Logan.");
		} else {
			System.out.println("Final score: $"+Money.money);
		}
		System.out.println("\n\n");
	}
}


src/stock/b2kan/com/Money.java
package stock.b2kan.com;

import java.util.Scanner;

public class Money {
	private static String yesno	= "";
	public static double money		= 100.0;
	public static double debt		= 0.0;
	public static boolean borrowed	= false, lastChance = false;
	private static int input		= 0;
	private static Scanner inputScan= new Scanner(System.in);
	
	public static void loanShark() {
		System.out.println("\n\nLoan Shark\nHow much would you like to borrow from Logan the loan shark? (intrest rate of 5% per day)\n");
		System.out.print("$");
		input	= inputScan.nextInt();
		System.out.println("Are you sure you wish to borrow $"+input+"? Intrest will be $"+(input * .05)+" per day. (y/n)");
		yesno	= inputScan.next();
		
		if(yesno.equals("y") || yesno.equals("Y") || yesno.equals("yes")) {
			debt		= input;
			money		= money + input;
			borrowed	= true;
			Main.news		= "You have been granted the money. Logan will be visiting in 5 days to get his money back.";
			Main.daysLeft	= 5;
		} else {
			System.out.println("Logan walked.");
		}
	}

	public static double fluctuate(double fluctuation, String trait) {
		int chance	= Main.rand.nextInt(3);
		double random	= Main.rand.nextDouble();
		int[] change	= {0, 0, 0};
		
		if(trait.equals("STABLE")) {
			change[0] = 2;
			change[1] = 5;
			change[2] = 1;
		} else {
			change[0] = 3;
			change[1] = 7;
			change[2] = 3;
		}
		// 0 = Stable(ish), 1 = Positive shift, 2 = Negative shift.
		switch(chance) {
			case 0: fluctuation	= fluctuation + (random * change[0]) - change[0];
			case 1: fluctuation	= fluctuation + (random * change[1]);
			case 2: fluctuation	= fluctuation - (random * change[2]);
		}
		
		// If we're going through a financial crisis
		if(Main.gfc == true) {
			fluctuation	= fluctuation - (fluctuation * .8);
		}
		
		// If things turned out really bad for the company and their share price is bad, make sure the price can't go below $0
		if(fluctuation < 0.0) {
			fluctuation	= 0.0;
		}
		
		return (Math.round(fluctuation * 100.0) / 100.0);
	}
	
	public static boolean enoughFunds(int companyID, int quantity) {
		if(Money.money >= (Main.exchange[companyID-1] * quantity)) {
			return true;
		} else {
			return false;
		}
	}
	
	public static void buyStock() {
		int amount	= 0;
		double currentPrice = 0;
		System.out.println("\n\nWhich company would you like to buy shares in?\n" +
							"1. Derp Inc.\n" +
							"2. Herp Industries.\n" +
							"3. Foo Co.\n" +
							"4. Bar Pty Ltd.\n" +
							"5. Nanohard\n" +
							"6. Applesoft\n" +
							"7. Back.");
		input	= inputScan.nextInt();
		if(input == 7) {
			Main.menu();
		}
		System.out.println("How many?");
		amount	= inputScan.nextInt();
		
		if(input > 0 && input < 7) {
			if(Money.enoughFunds(input, amount) == true) {
				currentPrice	= Main.exchange[input-1];
				if(currentPrice == 0.0) {
					System.out.println("You can't buy stocks at $0.00");
					Game.sleep(500);
				} else {
					Main.inventory1[input-1]	= Main.inventory1[input-1] + amount;
					Main.inventory2[input-1]	= currentPrice;
					Money.money					= Money.money - (currentPrice * amount);
					System.out.println("Stocks purchased!");
				}
			} else {
				System.out.println("You don't have enough capital to buy these shares.");
			}
		}
	}
	
	public static void sellStock() {
		int amount	= 0;
		double oldPrice	= 0.0;
		double currentPrice = 0.0;
		System.out.println("\n\nWhich company would you like to sell shares from?\n" +
							"1. Derp Inc.		("+Main.inventory1[0]+")\n" +
							"2. Herp Industries.	("+Main.inventory1[1]+")\n" +
							"3. Foo Co.		("+Main.inventory1[2]+")\n" +
							"4. Bar Pty Ltd.		("+Main.inventory1[3]+")\n" +
							"5. Nanohard		("+Main.inventory1[4]+")\n" +
							"6. Applesoft		("+Main.inventory1[5]+")\n" +
							"7. Back.");
		input	= inputScan.nextInt();
		if(input == 7) {
			Main.menu();
		}
		System.out.println("How many?");
		amount	= inputScan.nextInt();
		
		if(amount > Main.inventory1[input-1]) {
			System.out.println("You only have "+Main.inventory1[input-1]+" shares in that company.");
			Game.sleep(500);
			sellStock();
		} else {
			oldPrice		= Main.inventory2[input-1];
			currentPrice	= Main.exchange[input-1];
			Main.inventory1[input-1]	= Main.inventory1[input-1] - amount;
			Money.money	= Money.money + (amount * currentPrice);
			
			System.out.println("Shares sold! Profit made: $"+((currentPrice - oldPrice)*amount));
		}
	}
}


src/stock/b2kan/com/Main.java
/*
 * Stock Market Game v0.2
 * 
 * Protected under a Creative Commons Attribution 3.0 Unported License,
 * Tom Penney 2012
 * 
 */

package stock.b2kan.com;

import java.util.Random;
import java.util.Scanner;
import java.lang.Math;

public class Main {
	public static Random rand		= new Random();
	public static Scanner inputScan	= new Scanner(System.in);
	public static Scanner inputScan2= new Scanner(System.in);
	public static int input, day, daysLeft = 0, count = 0, gfcCount = 0;
	public static String version	= "0.5";
	public static int[] inventory1	= {0, 0, 0, 0, 0, 0};
	public static double[] inventory2	= {0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
	public static boolean gfc		= false;
	public static int rumor			= -1;
	public static String thisRumor, oldRumor, rumorResult = "", news = "";
	public static String[] rumors	= {"Derp Inc have hired a new CEO, share prices are expected to drop.", "Charity commends Herp Industries for their most recent work, share prices set to rise.", "Foo Co has been rumored to be carelessly dumping toxic waste into a nearby river, prices set to drop.", "Bar Pty Ltd released smear campaign against Foo Co, Foo Co prices expected to fall.", "Experts predict Foo Co share prices are going to rise.", "There are rumors that Bar Pty Ltd is bankrupt and will pull out of the stock exchange. Share prices are expected to drop severely."};
	public static String[] listed	= {"Derp Inc", "Herp Industries", "Foo Corp", "Bar Pty Ltd", "Nanohard", "Applesoft"};
	public static double[] exchange	= {0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
	public static String[] traits	= {"STABLE", "UNSTABLE", "STABLE", "UNSTABLE", "STABLE", "UNSTABLE"};
	public static boolean barBankrupt= false;
	
	public static void main(String[] args) {
		init();
		//if(args.length == 1) {
			//if(args[0].equals("showgui")) {
				GUI.showGUI();
			//}
		//}
		menu();
	}
	
	public static void menu() {
		System.out.println("\n\nStock Market Game v"+version+"\n" +
							"You have $"+Money.money+" in the bank.");
		if(Money.borrowed == true) { System.out.println("You are $"+Money.debt+" in debt"); }
		Game.showNews();
		System.out.println("\n1. Stock Prices.\n" +
							"2. Buy Stocks.\n" +
							"3. Sell Stocks.\n" +
							"4. My Shares.\n" +
							"5. Next Day.\n" +
							"6. Loan Shark.\n" +
							"7. Exit.");
		if(inputScan.hasNextInt() == true) {
			input	= inputScan.nextInt();
		} else {
			inputScan.next();
		}
		
		switch(input) {
			case 1: stockPrices(); menu();
			case 2: Money.buyStock(); menu();
			case 3: Money.sellStock(); menu();
			case 4: myShares(); menu();
			case 5: nextDay(); stockPrices(); menu();
			case 6: Money.loanShark(); menu();
			case 7: System.exit(0);
			case 8: next100(); menu();
			case 9: while(count <= 100) { Game.rollForRumor(); count++; } count = 0; menu();
			case 100: Money.money = Money.money + 500; menu();
			default: System.out.println("Please enter a valid choice."); Game.sleep(500); menu();
		}
	}
	
	private static void init() {
		// Make all the prices a random number so we have something to from.
		for(int i=0; i < 6; i++) {
			exchange[i]	= rand.nextInt(40) + 1;
			if(i == 5 || i == 6) {
				exchange[i]	= rand.nextInt(180) + 20;
			}
		}
		day		= 1;
	}

	private static void stockPrices() {
		for(int i=0; i < listed.length; i++) {
			System.out.println(listed[i]+"	"+"$"+exchange[i]);
		}
	}
	
	public static void myShares() {
		int count	= 0;
		System.out.println("\nMy shares:");
		for(int i=0; i < inventory1.length; i++) {
			if(inventory1[i] != 0) {
				System.out.println(listed[i]+"	"+inventory1[i]+" shares at $"+inventory2[i]+" each.");
				count++;
			}
		}
		if(count == 0) {
			System.out.println("You do not have any shares yet.");
		}
	}

	public static void nextDay() {
		for(int i=0; i < exchange.length; i++) {
			exchange[i]	= Money.fluctuate(exchange[i], traits[i]);
		}
		
		// If there is a rumor
		if(rumor != -1) {
			// Chance if the rumor was true or false
			if(rand.nextInt(3) != 1) { // If the rumor is true
				if(gfc == false) {
					switch(rumor) {
						case 0: exchange[0]	= exchange[0] - (exchange[0] * .3); exchange[0]	= Math.round(exchange[0] * 100.0) / 100.0;
						case 1: exchange[1]	= exchange[1] + (exchange[1] * .2); exchange[1]	= Math.round(exchange[1] * 100.0) / 100.0;
						case 2: exchange[2]	= exchange[2] - (exchange[2] * .4); exchange[2]	= Math.round(exchange[2] * 100.0) / 100.0;
						case 3: exchange[2]	= exchange[2] - (exchange[2] * .6); exchange[2]	= Math.round(exchange[2] * 100.0) / 100.0;
					}
					thisRumor	= "The rumor was true!";
				}
				rumor		= -1; // Set to -2 because the rumor "it was true" doesn't deserve a postive number
			} else { // If the rumor is false
				thisRumor	= "The rumor was false!";
				rumor		= -1; // Set to -2 for the same reason as above ^
			}
		} else {
			if(thisRumor == "") {
				Game.rollForGfc();
				if(gfc == false) {
					Game.rollForRumor();
					Game.rollForBuyBack();
				} else {
					gfcCount	= 3;
				}
			} else {
				rumor = -1;
				thisRumor	= "";
				rumorResult	= "";
			}
		}
		
		if(daysLeft != 0) {
			daysLeft--;
			Money.debt	= Money.debt + (Money.debt * .05);
		}
		
		if(daysLeft == 1) {
			news	= "Logan will be stopping by tomorrow, make sure you have his money!";
		}
		if(Money.borrowed == true && daysLeft == 0) {
			if(Money.money < Money.debt) {
				news	= "You don't have enough money to pay back Logan. You have one more day to pay him back.";
				Money.lastChance	= true;
			} else {
				Money.lastChance	= false;
				Money.borrowed	= false;
				Money.money		= Money.money - Money.debt;
			}
			if(Money.lastChance == true) {
				if(Money.money < Money.debt) {
					Game.gameOver(1);
				}
			}
		}
		
		if(barBankrupt == true) {
			exchange[3]	= 0.0;
		}
		
		if(gfc == true) {
			gfcCount--;
		}
		
		if(gfc == true && gfcCount == 0) {
			gfc	= false;
		}
	}
	
	public static void next100() {
		int run	= 0;
		for(int i=0; i<exchange.length; i++) {
			System.out.println("\n");
			while(run <= 100) {
				exchange[i]	= Money.fluctuate(exchange[i], traits[i]);
				System.out.println(exchange[i]);
				run++;
			} run	= 0;
		}
	}
}
